ads/imedia.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, computeInMasterFrame, validateData} from '../3p/3p';
+
+/* global im:false */
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function imedia(global, data) {
+  validateData(data, ['id','positions']);
+  let positions = null;
+  if (data.positions) {
```

Comment:
If I remove this condition, gulp check-types won't pass. And I will get this error:
```
Error compiling ./3p/integration.js
Command failed: java -jar -XX:+TieredCompilation build-system/runner/dist/runner.jar --flagfile="/var/folders/d0/h4nrdlvn4l7dmcs4534jc5rsm20kq1/T/662c3cb0-6aea-4c67-b90a-f1c4ece79d15" --compilation_level=SIMPLE_OPTIMIZATIONS --assume_function_wrapper=true --language_in=ECMASCRIPT6 --language_out=ECMASCRIPT5 --rewrite_polyfills=false --externs=build-system/amp.extern.js --externs=third_party/closure-compiler/externs/intersection_observer.js --externs=third_party/closure-compiler/externs/shadow_dom.js --externs=third_party/closure-compiler/externs/web_animations.js --externs=ads/ads.extern.js --externs=build-system/amp.oti.extern.js --js_module_root=node_modules/ --js_module_root=build/patched-module/ --js_module_root=build/fake-module/ --entry_point=./3p/integration.js --process_common_js_modules=true --only_closure_dependencies=true --output_wrapper=(function(){%output%})();
//# sourceMappingURL=integration-check-types.js.map
 --create_source_map=build/cc/_3p_integration.js.map --source_map_location_mapping=|http://localhost:8000/ --warning_level=DEFAULT --jscomp_off=unknownDefines --define=TYPECHECK_ONLY=true --hide_warnings_for=third_party/caja/ --hide_warnings_for=third_party/closure-library/sha384-generated.js --hide_warnings_for=third_party/d3/ --hide_warnings_for=third_party/vega/ --hide_warnings_for=third_party/webcomponentsjs/ --hide_warnings_for=node_modules/ --hide_warnings_for=build/patched-module/ --hide_warnings_for=3p/environment.js --jscomp_error=checkTypes --jscomp_error=accessControls --jscomp_error=const --jscomp_error=constantProperty --jscomp_error=globalThis --conformance_configs=build-system/conformance-config.textproto --js_output_file="build/cc/_3p_integration.js"
ads/imedia.js:28: ERROR - Property positions never defined on Object
  positions = JSON.parse(data.positions);
                              ^

1 error(s), 0 warning(s), 86.98026975810946% typed`
```